quantities/serialization/QuantityFactory.cs

```diff
@@ -0,0 +1,86 @@
+using System.Collections.Concurrent;
+using Quantities.Dimensions;
+using Quantities.Measures;
+
+namespace Quantities.Serialization;
+
+file static class Inject
+{
+    public static IInject Fractional { get; } = new FractionInjector();
+    public static IInject Multiplicative { get; } = new ProductInjector();
+    public static IInject Square { get; } = new PowerInjector<Square>();
+    public static IInject Cubic { get; } = new PowerInjector<Cubic>();
+    public static IInject Scalar { get; } = new ScalarInjector();
+}
+
+public readonly struct QuantityFactory<TQuantity>
+    where TQuantity : IFactory<TQuantity>
+{
+    private static readonly Type typeofQuantity = typeof(TQuantity);
+    private static readonly ConcurrentDictionary<Int32, IBuilder> complexCache = new();
```

Comment:
why not use the fully localized (namespace+quantityname) for this cache? It would make inspection a bit easier.